### PR TITLE
Collectors: Stabilise bear order

### PR DESCRIPTION
**Pull request by jayvdb**
Fixes https://github.com/coala/coala/issues/4541

### CODE REVIEW

File: coalib/collecting/Collectors.py [hunk 1]
```diff
@@ -126,9 +126,12 @@ def icollect_bears(bear_dir_glob, bear_globs, kinds, log_printer):
         # pass this later to iglob, we need to escape this.
         bear_dir = glob_escape(bear_dir)
         for bear_glob in bear_globs:
-            for matching_file in iglob(
-                    os.path.join(bear_dir, bear_glob + '.py')):
+            matching_files = iglob(
+                    os.path.join(bear_dir, bear_glob + '.py'))
 
+            matching_files = sorted(matching_files)
```
Comment (Makman2) [inline coalib/collecting/Collectors.py]: maybe just
```
for matching_file in sorted(iglob(
        os.path.join(...))):
```
Saves some modifications :+1:

File: tests/settings/ConfigurationGatheringTest.py [hunk 2]
```diff
@@ -369,13 +373,20 @@ def test_get_filtered_bears(self):
...
-        self.assertEqual(str(local_bears['cli'][1]),
-                         "<class 'LineCountTestBear.LineCountTestBear'>")
+        expected = [
+            "<class 'JavaTestBear.JavaTestBear'>",
+            "<class 'LineCountTestBear.LineCountTestBear'>",
+        ]
+
+        self.assertEqual(
+            [str(bear) for bear in local_bears['cli']],
+            expected)
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: if you like you can inline `expected`, up to you :+1:

Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: Separate commit for this one? As this is effectively a rewrite^^

File: coalib/collecting/Collectors.py (hunk 1 above)
Comment (jayvdb) [inline coalib/collecting/Collectors.py]: No change in the number of lines modified (2), as you still need to add the trailing `)`.
My version allows for more complicated sorting to be added later if needed.

File: tests/settings/ConfigurationGatheringTest.py (hunk 2 above)
Comment (jayvdb) [inline tests/settings/ConfigurationGatheringTest.py]: I didnt notice it was a rewrite; there were a few iterations before the order became the same as the previous test order.

I think I'll just revert, as this doesnt add much.

File: coalib/collecting/Collectors.py [hunk 3]
```diff
@@ -126,9 +126,12 @@ def icollect_bears(bear_dir_glob, bear_globs, kinds, log_printer):
         # pass this later to iglob, we need to escape this.
         bear_dir = glob_escape(bear_dir)
         for bear_glob in bear_globs:
-            for matching_file in iglob(
-                    os.path.join(bear_dir, bear_glob + '.py')):
+            matching_files = iglob(
+                    os.path.join(bear_dir, bear_glob + '.py'))
```
Comment (Makman2) [inline coalib/collecting/Collectors.py]: just 4 spaces indentation :+1:

Comment (Makman2) [inline coalib/collecting/Collectors.py]: -->
```
matching_files = iglob(
    os.path.join(bear_dir, bear_glob + '.py'))
```

**Comment by Makman2** [comment]
ack 776dbbd

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**